modules/reactor/include/meshgenerators/ReactorGeometryMeshBuilderBase.h

```diff
@@ -0,0 +1,92 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MeshGenerator.h"
+#include "libmesh/elem.h"
+
+/**
+ * A base class that contains common members for Reactor Geometry Mesh Builder mesh generators.
+ */
+class ReactorGeometryMeshBuilderBase : public MeshGenerator
+{
+public:
+  static InputParameters validParams();
+
+  ReactorGeometryMeshBuilderBase(const InputParameters & parameters);
+
+  /**
+   * Checks input mesh and returns max(block ID) + 1, which represents
+   * a block ID in the mesh that is not currently in use
+   * @param input mesh over which to compute next free block id
+   */
+  SubdomainID nextFreeId(MeshBase & input_mesh);
```

Comment:
move this to mesh utils. getNextFreeSubdomainID

freeId is not descriptive enough

Reply:
This is now moved to mose mesh util and the code in lowerDBlockFromSidesetGenerator also calls this function